driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/ClientSession.java

```diff
@@ -35,6 +35,17 @@
      */
     boolean hasActiveTransaction();
 
+    /**
+     *  Notify the client session that a message has been sent.
+     *  <p>
+     *      For internal use only
+     *  </p>
+     *
+     * @return true if this is the first message sent, false otherwise
+     * @since 4.0
+     */
+    boolean notifyMessageSent();
```

Comment:
Noticed this missing method in the reactive API, and had to add it to implement the sync adapter.